src/collection-facets/more-facets-content.ts

```diff
@@ -0,0 +1,371 @@
+                      .identifier=${displayText}
+                      placeholder="-"
+                    ></async-collection-name>`}
+              </div>
+              <div class="facet-count">${facet.doc_count}</div>
+            </label>
+          </li>
+        `;
+      })}
+    </ul>`;
+  }
+
+  private facetClicked(e: Event) {
+    const { selectedFacets } = this;
+
+    const target = e.target as HTMLInputElement;
+    const { checked, value } = target;
+
+    let newFacets: SelectedFacets;
+    if (selectedFacets) {
+      newFacets = {
+        ...selectedFacets,
+      };
+    } else {
+      newFacets = defaultSelectedFacets;
+    }
+
+    if (checked) {
+      newFacets[this.facetKey as keyof typeof newFacets][value] = 'selected';
+    } else {
+      delete newFacets[this.facetKey as keyof typeof newFacets][value];
+    }
+
+    this.selectedFacets = newFacets;
+  }
+
+  private get loaderTemplate() {
+    return this.facetsLoading
+      ? html`<div class="facets-loader">
+          <ia-activity-indicator .mode="processing"></ia-activity-indicator>
+        </div>`
+      : '';
+  }
+
+  // render pagination if more then 1 page
+  private get facetsPaginationTemplate() {
+    return this.paginationSize > 1
+      ? html`<more-facets-pagination
+          .size=${this.paginationSize}
+          @pageNumberClicked=${this.pageNumberClicked}
+        ></more-facets-pagination>`
+      : nothing;
+  }
+
+  private get facetsContentTemplate() {
+    return html`
+      <div class="facets-content">${this.getMoreFacetsTemplate}</div>
+      ${this.paginationSize > 0
+        ? html`${this.facetsPaginationTemplate}
+            <div class="footer">
+              <button
+                class="btn btn-cancel"
+                type="button"
+                @click=${this.cancelClick}
+              >
+                Cancel
+              </button>
+              <button
+                class="btn btn-submit"
+                type="button"
+                @click=${this.applySearchFacetsClicked}
+              >
+                Apply filters
+              </button>
+            </div>`
+        : html`No result found. please try again later.`}
+    `;
+  }
+
+  render() {
+    return html`
+      <div id="more-facets-page">
+        <form>
+          ${this.facetsLoading
+            ? this.loaderTemplate
+            : this.facetsContentTemplate}
+        </form>
+      </div>
+    `;
+  }
+
+  private applySearchFacetsClicked() {
+    const event = new CustomEvent<SelectedFacets>('facetsChanged', {
+      detail: this.selectedFacets,
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+    this.modalManager?.closeModal();
+  }
+
+  private cancelClick() {
+    this.modalManager?.closeModal();
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      #more-facets-page {
+        margin-bottom: 2rem;
+      }
+      .modal-content {
+        background-color: #fefefe;
```

Comment:
QUESTION: Again, here and elsewhere. Should colors not be coming in via some kind of global CSS that reflects a styleguide?